### PR TITLE
[SYTO-1169] Added Secondary Nav to Color Page

### PR DESCRIPTION
**Pull request by krlospelaez**
To test:

- Go to Color page
- Verify it contains a left rail menu
- Verify all links are working correctly.

### CODE REVIEW

File: src/site/color/index.html [hunk 1]
```diff
@@ -12,11 +12,27 @@ <h1 class="mt3_heading--h1--2"> Color Palette </h1>
   </div>
 </div>
 
-<div class="mt3_bgcolor--gray5 mt3_row-gut-4 divider">&nbsp;</div>
```
Comment (hkekeli) [inline src/site/color/index.html]: You need this part.

File: src/styles/theme/socialbrands.html [hunk 2]
```diff
@@ -1,10 +1,10 @@
-<div class="pagewrapper mt3_row">
-  <div class="mt3_col-lg-3 mt3_col-gut">
+<div class="mt3_col-lg-offset-4 mt3_row">
+  <div class="mt3_col-gut">
     <h3 class="mt3_heading--h4--2">Social Brands</h3>
     <p class="mt3_subh4">Whenever possible,social icons should be displayed in their native brand colors, or in white on a native background color field.</p>
   </div>
 
-  <div class="mt3_col-lg-8 mt3_col-sm-12 mt3_col-lg-offset-1">
+  <div class="">
```
Comment (hkekeli) [inline src/styles/theme/socialbrands.html]: This is probably not needed anymore...

File: src/site/nav/SecondaryNav.js [hunk 3]
```diff
@@ -31,10 +57,29 @@ export default class SecondaryNav extends Component {
...
-  renderSecondaryNavigation: function(el){
-    DOMrender(<SecondaryNav />, el);
+  renderSecondaryNavigation: function(el) {
+    if (el.classList.contains('secondary-nav-sticky')) {
+      ReactDOM.render(<SecondaryNav sticky />, el);
+    } else {
+      ReactDOM.render(<SecondaryNav />, el);
+    }
   }
 }
```
Comment (hkekeli) [inline src/site/nav/SecondaryNav.js]: Great!

File: src/styles/theme/colordoc.html [hunk 4]
```diff
@@ -1,13 +1,13 @@
-<div class="pagewrapper mt3_row">
-  <div class="mt3_col-lg-3 mt3_col-gut">
+<div class="mt3_col-lg-offset-4 mt3_row">
+  <div class="mt3_col-gut">
```
Comment (hkekeli) [inline src/styles/theme/colordoc.html]: You can move this classname to the div above and remove this one

File: src/styles/theme/primary.html [hunk 5]
```diff
@@ -1,5 +1,5 @@
-<div class="pagewrapper mt3_row">
-  <div class="mt3_col-lg-3 mt3_col-gut">
+<div class="mt3_col-lg-offset-4 mt3_row">
+  <div class="mt3_col-gut">
```
Comment (hkekeli) [inline src/styles/theme/primary.html]: here too

File: src/site/color/index.html [hunk 6]
```diff
@@ -12,11 +12,27 @@ <h1 class="mt3_heading--h1--2"> Color Palette </h1>
...
+  <a class="anchor-link" name="primary"></a>
+  {% include "src/styles/theme/primary.html" %}
+
+  <a class="anchor-link" name="neutrals"></a>
+  {% include "src/styles/theme/neutrals.html" %}
+
+  <a class="anchor-link" name="miscellaneous"></a>
+  {% include "src/styles/theme/miscellaneous.html" %}
+
+  <a class="anchor-link" name="socialbrands"></a>
```
Comment (hkekeli) [inline src/site/color/index.html]: We should be consistent with the formatting on these anchor names. 

Ex: 
<a name="video-playlist">
<a name="social-brands">



<a name="video-playlist">

Comment (pmourelle) [inline src/site/color/index.html]: Seems pretty consistent to me having same name attr as file name.

File: src/site/nav/SecondaryNav.js [hunk 7]
```diff
@@ -1,8 +1,34 @@
...
+import zip from 'lodash/zip';
+import throttle from 'lodash/throttle';
+
+const SECONDARY_NAV_FIXED_TOP = 90; // px
 
 export default class SecondaryNav extends Component {
+  constructor(props) {
+    super(props);
+
+    this.setStickyMenu = this.setStickyMenu.bind(this);
```
Comment (pmourelle) [inline src/site/nav/SecondaryNav.js]: Use the `::` form please

Comment (pmourelle) [inline src/site/nav/SecondaryNav.js]: Also, if `props` isn't used, you may omit it.

File: src/site/nav/SecondaryNav.js [hunk 8]
```diff
@@ -1,8 +1,34 @@
...
+    this.handleStickyMenu = throttle(this.setStickyMenu, 50);
+  }
+
+  componentDidMount() {
+    if(!this.props.sticky) {
```
Comment (pmourelle) [inline src/site/nav/SecondaryNav.js]: `if (`

File: src/site/nav/SecondaryNav.js [hunk 9]
```diff
@@ -31,10 +57,29 @@ export default class SecondaryNav extends Component {
       </div>
     )
   }
+
+  setStickyMenu() {
+    const windowTop = window.pageYOffset;
+    if (windowTop + SECONDARY_NAV_FIXED_TOP > this.secondaryNavTop) {
+      Object.assign(this.secondaryNavEl.style, {
```
Comment (pmourelle) [inline src/site/nav/SecondaryNav.js]:
```
this.secondaryNavEl.style = {
  ...this.secondaryNavEl.style,
  position: 'fixed',
  top: `${SECONDARY_NAV_FIXED_TOP}px`
}
```

Also, I'm not sure about using all caps snake_case for constants...

File: src/site/nav/SecondaryNav.js [hunk 10]
```diff
@@ -31,10 +57,29 @@ export default class SecondaryNav extends Component {
...
+        position: 'fixed',
+        top: SECONDARY_NAV_FIXED_TOP + 'px'
+      });
+    } else {
+      Object.assign(this.secondaryNavEl.style, {
```
Comment (pmourelle) [inline src/site/nav/SecondaryNav.js]: Same as above. ES6 destructuring ftw!

File: src/site/color/index.html (hunk 6 above)
Comment (hkekeli) [inline src/site/color/index.html]: @pmourelle Yea I was referring to the way they were formatted on this ticket: https://jira.natgeo.com/browse/SYTO-859

At least that's the pattern they're using here: http://www.nationalgeographic.com/photography/photo-of-the-day/

File: src/site/color/index.html (hunk 1 above)
Comment (hkekeli) [inline src/site/color/index.html]: @krlospelaez this is the divider that separates the intro from the rest of the page ^

Comment (hkekeli) [inline src/site/color/index.html]: Also seeing this: 

divider bug

File: src/site/color/index.html [hunk 11]
```diff
@@ -12,11 +12,28 @@ <h1 class="mt3_heading--h1--2"> Color Palette </h1>
...
-<div class="mt3_bgcolor--gray5 mt3_row-gut-4 divider">&nbsp;</div>
-
-{% include "src/styles/theme/colordoc.html" %}
-<div class="mt3_bgcolor--gray5 mt3_row-gut-4 divider">&nbsp;</div>
-{% include "src/styles/theme/primary.html" %}
-{% include "src/styles/theme/neutrals.html" %}
-{% include "src/styles/theme/miscellaneous.html" %}
-{% include "src/styles/theme/socialbrands.html" %}
+<div class="pagewrapper mt3_col-gut">
+    <div id="secondaryNav" class="secondary-nav-sticky"></div>
```
Comment (hkekeli) [inline src/site/color/index.html]: Indent

**Pull request merged**